service/internal/endpoint-discovery/cache.go

```diff
@@ -0,0 +1,98 @@
+package endpointdiscovery
+
+import (
+	"sync"
+	"sync/atomic"
+)
+
+// DefaultCacheSize is the default endpoint discovery cache size that should be
+// used to construct endpoint cache, if not provided by customer
+const DefaultEndpointCacheSize int64 = 10
```

Comment:
This is referred by the client to build a new cache. 

Not much opinion on this, as it is hard-coded any way. Keeping it here let's us keep value in go code, instead of some constant in java code